lib/Demangling/Demangler.cpp

```diff
@@ -774,12 +774,16 @@ NodePointer Demangler::demangleType(StringRef MangledName,
         std::function<SymbolicReferenceResolver_t> Resolver) {
   DemangleInitRAII state(*this, MangledName, std::move(Resolver));
 
-  parseAndPushNodes();
+  if (!parseAndPushNodes())
+    return nullptr;
 
-  if (NodePointer Result = popNode())
-    return Result;
+  NodePointer Result = popNode();
+
+  // The result is only valid if it was the only node on the stack.
+  if (popNode())
+    return nullptr;
 
-  return createNode(Node::Kind::Suffix, Text);
+  return Result;
```

Comment:
Is there harm to keeping the `Suffix` production? The runtime demangler should never accept it as a valid mangling, but it's helpful when demangling generated symbol names in binaries. Could it be kept as an option?

Reply:
Yeah, it can also be helpful when debugging mangling errors, since it generally points at exactly what you messed up.

Reply:
This change only affects `demangleType` (the runtime demangler). Nothing changes for demangling symbols

Reply:
> debugging mangling errors

Pro tip: `Demangler::dump()` prints the internal state of the demangler which is very useful for debugging mangling problems